CMakeLists.txt

```diff
@@ -49,38 +49,75 @@
-# For instance:
 #
-#   cmake -DLOMSE_PKG_CONFIG_INSTALL=/usr/lib/x86_64-linux-gnu/pkgconfig  ...
+# a) Installation options
+# --------------------------------------------
+#
+# MAKE_INSTALL_PREFIX  (Default: "/usr/local")
+# 	The default installation prefix is "/usr/local". Therefore, libraries
+#	(binaries) will be installed at "/usr/local/lib" and include files at
+# 	"/usr/local/include/lomse". You can change the default location 
+# 	by specifying option CMAKE_INSTALL_PREFIX. For instance:
+#   	cmake -DCMAKE_INSTALL_PREFIX=/usr  ....
 #
+# LOMSE_PKG_CONFIG_INSTALL  (Default "/usr/lib/pkgconfig")
+# 	As part of the Lomse instalation process, a pkg-config file, named
+#	liblomse.pc, is installed. The default location is "/usr/lib/pkgconfig".
+#	You can change this default location by using the option
+#	-DLOMSE_PKG_CONFIG_INSTALL=/new/pkgconfig/path. For instance:
+#   	cmake -DLOMSE_PKG_CONFIG_INSTALL=/usr/lib/x86_64-linux-gnu/pkgconfig  ....
 #
-# Additional build options (ON / OFF values):
+#
+# b) Debug options (ON / OFF values):
 # --------------------------------------------
-# Values can be changed directly in the CMake GUI or through
-# the command line by prefixing a variable's name with '-D':
-# i.e.:    cmake -DLOMSE_DEBUG=ON  ...
 #
 # LOMSE_ENABLE_DEBUG_LOGS   (Default value: OFF)
-#       Enable debug logs (performance loss). Doesn't require a debug build.
+#	Enable debug logs (performance loss). Doesn't require a debug build.
 #
 # LOMSE_DEBUG   (Default value: OFF)      
-#       Force to create a debug build, with debug symbols.
+#	Force to create a debug build, with debug symbols.
+#
+#
+# c) Options to reduce dependencies from other libraries (ON / OFF values):
+# --------------------------------------------------------------------------
+#
+# If your application will not deal with compressed files or with scores
+# containing png images, it is possible to reduce dependencies from third
+# party librraries by removing the code that deals with compressed files
+# or with PNG images.
+#
+# LOMSE_ENABLE_PNG   (Default value: ON)
+#	Build with support for PNG images. This requires libpng library.
+#	By setting this to OFF Lomse will ignore any png image embedded in the
+#	documents. But Lomse will not fail in those cases, just will log error
+#	messages. For instance:
+#		cmake -G "Unix Makefiles" -DLOMSE_ENABLE_PNG=OFF ....
 #
+# LOMSE_ENABLE_COMPRESSION   (Default value: ON)
+#	Build with support for compressed files. This requires zlib library.
+#	By setting this to OFF Lomse will not be able to deal with compressed files,
+#	such as MusicXML files in compressed format, or with png images embedded in
+#	the documents. But Lomse will not fail in those cases, just will log error
+#	messages. For instance:
+#		cmake -G "Unix Makefiles" -DLOMSE_ENABLE_COMPRESSION=OFF ....
+#
+#	**Important**: deactivating compression will also deactivate support for
+#	PNG images because libpng depends on zlib.
```

Comment:
The description differs from current implementation: trying to disable compression without also disabling png has no effect. If png is active then compression is automatically activated.

If you want it the way you describe it that can be easily achieved but requires small modifications to CMake configuration.

Reply:
Thanks. I will fix the description.

I've tested the PR and it works as it is now. But I'm considering to remove the crash (throw) when the library has been built without zlib support and it is requested to open a compressed file. I think it is better not to throw an exception but to log an error message and fail opening the file (this will report a 'not supported format' error, and will not cause a crash in the user application).

Reply:
I saw exceptions in other places and thought it's the way to report errors. It's certainly not good to throw exception in one case but report errors via empty objects or using error codes in others. Consistency of behaviour is important.

Please make it the way you like as you of course know design choices of your project better.

Reply:
I started with this time ago. At start, I always threw exceptions. But after starting to use the library in my application I realize that when feasible and convenient it is better to log errors and do not cause crashes. There are still many points in which the library throw exceptions, but I didn't have time for reconsidering them and do not cause problems in applications. Most of them are at places in which there must have been a very bad failure in logic or program design.